lib/src/install.rs

```diff
@@ -777,27 +778,53 @@ impl RootSetup {
     }
 }
 
+pub(crate) enum SELinuxFinalState {
+    /// Host and target both have SELinux, but user forced it off for target
+    ForceTargetDisabled,
+    /// Host and target both have SELinux
+    Enabled(Option<crate::lsm::SetEnforceGuard>),
```

Comment:
This throws a `dead_code` warning with rust 1.77, see: https://github.com/rust-lang/rust/issues/122833

Can either `#[allow(dead_code)]` here, or alternatively I've verified you can change it to the struct-like variant and hint at it with underscore, so something like this works too:

```rust
Enabled { _guard: Option<crate::lsm::SetEnforcedGuard> }
```

Reply:
Ah thanks, yeah my toolbox here is out of date.

Hmm...so I decided to fix this in a different way because this "execute code by side effect on `Drop`" was always a code smell, so I've changed it so we actually do process it in the normal path.

Reply:
This all said, from time to time I've thought about writing a crate which basically forks off a helper child process that acts as "cleanup on parent process exit" for exactly things like temporary directories or other "external global" state - and it'd work really well for this; we'd rerun the `setenforce 1` even we panic'd etc.  (We use `panic=abort` here)